### DISCUSSION TITLE
[textadept-lsp]: typescript-language-server v4 not working

### DISCUSSION THREAD

**Discussion by daron81** [orbitalquark/textadept#502] [Q&A]
Hello,

I just noticed that, with the following config:
`local lsp = require("lsp")
lsp.server_commands.javascript = 'typescript-language-server --stdio'
lsp.server_commands.typescript = 'typescript-language-server --stdio'`

I could not get typescript-language-server v4 working, but v3 is working fine. Any ideas? Ty.

**Comment by orbitalquark** [comment]
Try putting `require('lsp').log_rpc = true` in your ~/.textadept/init.lua file, load a file that starts an LSP server, then go to Tools > Language Server > Show Log. Perhaps that will help shed light on the issue. Feel free to post the contents of the log for me to take a look.

Ideally you'd have logs for both v3 (working) and v4 (not working) in order to compare the two.

**Comment by daron81** [comment]
Sure thing. I've attached the logs for both versions. Thanks.

[tls-ls-log-v3.txt](https://github.com/orbitalquark/textadept/files/14316775/tls-ls-log-v3.txt)
[tls-ls-log-v4.txt](https://github.com/orbitalquark/textadept/files/14316776/tls-ls-log-v4.txt)

**Reply by orbitalquark** [reply]
These logs appear to be identical, so unfortunately I cannot see what's going wrong with v3 vs v4 :(

**Reply by daron81** [reply]
My apologies. I should have verified this.

Here is another set, with typescript-language-server's log-level bumped up. Hope these prove to be more helpful.

[ts-lsp-v3.txt](https://github.com/orbitalquark/textadept/files/14335526/ts-lsp-v3.txt)
[ts-lsp-v4.txt](https://github.com/orbitalquark/textadept/files/14335527/ts-lsp-v4.txt)

**Reply by orbitalquark** [reply]
So it looks like v4 is sending a "typescriptVersion" notification to the LSP, and Textadept is ignoring it, which is fine by itself. I don't know if this notification blocks up the I/O channel, though. After starting v4, when you try and do something like attempt a completion, nothing happens right? I'm not seeing any outgoing requests from Textadept to the v4 server after it sent the "typescriptVersion" notification.

**Reply by daron81** [reply]
Actually, it seems as though the auto-completion is working. It is the diagnostics that are not showing up, which is why I noticed it in the first place.

**Reply by orbitalquark** [reply]
I've committed a change that should notify servers that diagnostics are supported: https://github.com/orbitalquark/textadept-lsp/commit/aaba5bc93b6d5de39a04b9b5ec0fbaa8cf9e347c. Perhaps v4 is checking before sending. The latest release zip should now have this change. Hopefully it works for you.

**Reply by daron81** [reply]
Yes, that seems to have fixed it. Thank you!
